src/DomainInterface.php

```diff
@@ -99,4 +99,11 @@ public function toUnicode();
      * from the right-most label to the left-most label.
      */
     public function getIterator();
+    
+    /**
+     * return true if domain contains deviation characters.
+     * @see http://unicode.org/reports/tr46/#Transition_Considerations
+     * @return bool
+     **/
+    public function isTransitionalDifferent(): bool;
```

Comment:
Adding this method to the interface should be postponed for the next major release because you are modifying the public API of the interface if someone already has this method implemented it's BC break for him.